src/ogc/openSearch/OpenSearchService.js

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright 2015-2018 WorldWind Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * @exports OpenSearchService
+ */
+
+define([
+        '../../error/ArgumentError',
+        '../../util/Logger',
+        './responseFormats/atomParser/OpenSearchAtomParser',
+        './OpenSearchConstants',
+        './descriptionDocument/OpenSearchDescriptionDocument',
+        './responseFormats/OpenSearchParserRegistry',
+        './OpenSearchRequest',
+        './OpenSearchUtils',
+        '../../util/Promise'
+    ],
+    function (ArgumentError,
+              Logger,
+              OpenSearchAtomParser,
+              OpenSearchConstants,
+              OpenSearchDescriptionDocument,
+              OpenSearchParserRegistry,
+              OpenSearchRequest,
+              OpenSearchUtils,
+              Promise) {
+        'use strict';
+
+        /**
+         * Constructs a service for interacting with OpenSearch servers.
+         *
+         * @alias OpenSearchService
+         * @constructor
+         * @classdesc Provides a service for interacting with OpenSearch servers.
+         *
+         * The service exposes two methods:
+         *  - discover used to get the description document;
+         *  - search used for querying the search engine.
+         *
+         * By default, this service can handle Atom for EO and GeoJSON responses.
+         * Other types of response formats can by added using the registerParser method.
+         *
+         * @param {String} url The URL of the OpenSearch description document.
+         */
+        var OpenSearchService = function (url) {
+            this._url = url;
```

Comment:
The URL can be set here or as an option of the discover method. I think we should choose a single workflow, either a class is created per URL or the class is reusable and the URL given to the discover method. What do you think? Each option has further consequences in my opinion.